file_store.go

```diff
@@ -0,0 +1,275 @@
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package credentials
+
+import (
+	"bytes"
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+
+	"github.com/oras-project/oras-credentials-go/internal/ioutil"
+	"oras.land/oras-go/v2/registry/remote/auth"
+)
+
+// FileStore implements a credentials store using the docker configuration file
+// to keep the credentials in plain-text.
+type FileStore struct {
+	// DisableSave disable saving credentials in plain text.
+	// If DisableSave is set to true, Put() will return ErrPlainTextSaveDisabled.
+	DisableSave bool
+
+	// configPath is the path to the config file
+	configPath string
+	// content is the content of the config file
+	content map[string]json.RawMessage
+	// authsCache is a cache of the auths field of the config field
+	authsCache map[string]json.RawMessage
+	// rwLock is a read-write-lock
+	rwLock sync.RWMutex
+}
+
+const configFieldAuths = "auths"
+
+var (
+	// ErrInvalidConfigFormat is returned when the config format is invalid.
+	ErrInvalidConfigFormat = errors.New("invalid config format")
+	// ErrPlainTextSaveDisabled is returned by Put() when DisableSave is set
+	// to true.
+	ErrPlainTextSaveDisabled = errors.New("plain text save is disabled")
```

Comment:
Use `plaintext` instead of `plain text` since `plaintext` is a crypto term.

See: https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/p/plaintext-plain-text

Reply:
Got it!